libraries/AP_BattMonitor/AP_BattMonitor_SMBus_Generic.cpp

```diff
@@ -5,14 +5,21 @@
 #include "AP_BattMonitor_SMBus_Generic.h"
 #include <utility>
 
-uint8_t maxell_cell_ids[] = { 0x3f,  // cell 1
-                              0x3e,  // cell 2
-                              0x3d,  // cell 3
-                              0x3c,  // cell 4
-                              0x3b,  // cell 5
-                              0x3a}; // cell 6
+uint8_t smbus_cell_ids[] = { 0x3f,  // cell 1
+                             0x3e,  // cell 2
+                             0x3d,  // cell 3
+                             0x3c,  // cell 4
+                             0x3b,  // cell 5
+                             0x3a,  // cell 6
+                             0x39,  // cell 7
+                             0x38,  // cell 8
+                             0x37,  // cell 9
+                             0x36,  // cell 10
+                             0x35,  // cell 11
+                             0x34}; // cell 12
 
 #define SMBUS_READ_BLOCK_MAXIMUM_TRANSFER    0x20   // A Block Read or Write is allowed to transfer a maximum of 32 data bytes.
+#define SMBUS_CELL_COUNT_CHECK_TIMEOUT       15000000   // check cell count for up to 15 seconds
```

Comment:
Attach a unit to the define, or if you want it in seconds just make it be 15 here, and do the multiplication down where you need microseconds.

Reply:
ok txs.  Change to "15" and multiply it by 1,000,000 lower down.  Of course using micros is crazy but it's pre-existing and wasn't written by me.